main.js

```diff
@@ -1865,6 +1961,35 @@ function _select_Gateway() { // Check for Gateway used to
         },
     });
 
+    // expand recursive tasklist
+    for (var professionIdx in definedTask) {

```

Comment:
For me complexity of these loops is n^2 (unless I don't know something about the complexity of javascript 'in' implementation).
And that for n equal to approximately 10-20 (in worst case).

Compared to compexity n^3 with time-consuming GM_ operations.

I realized, no more "Pull requests".